processing/src/main/java/io/druid/query/groupby/GroupByQueryQueryToolChest.java

```diff
@@ -426,6 +427,11 @@ public Object apply(Row input)
               for (AggregatorFactory agg : aggs) {
```

Comment:
@a2l007 
```
final List<Object> retVal = Lists.newArrayListWithCapacity(1 + dims.size() + aggs.size());
```
a few lines above is not updated wrt getPostAggregatorSpecs. Similar problems in other classes, updated in this PR. Also I noticed that somewhere this sizing was buggy before this PR already.